cardano-api/test/Test/Cardano/Api/Crypto.hs

```diff
@@ -0,0 +1,216 @@
+        , testProperty "Sig"     $ prop_raw_serialise @(SigDSIGN v)
+                                                      rawSerialiseSigDSIGN
+                                                      rawDeserialiseSigDSIGN
+        ]
+
+      , testGroup "size"
+        [ testProperty "VerKey"  $ prop_size_serialise @(VerKeyDSIGN v)
+                                                       rawSerialiseVerKeyDSIGN
+                                                       (sizeVerKeyDSIGN (Proxy @ v))
+        , testProperty "SignKey" $ prop_size_serialise @(SignKeyDSIGN v)
+                                                       rawSerialiseSignKeyDSIGN
+                                                       (sizeSignKeyDSIGN (Proxy @ v))
+        , testProperty "Sig"     $ prop_size_serialise @(SigDSIGN v)
+                                                       rawSerialiseSigDSIGN
+                                                       (sizeSigDSIGN (Proxy @ v))
+        ]
+
+      , testGroup "direct CBOR"
+        [ testProperty "VerKey"  $ prop_cbor_with @(VerKeyDSIGN v)
+                                                  encodeVerKeyDSIGN
+                                                  decodeVerKeyDSIGN
+        , testProperty "SignKey" $ prop_cbor_with @(SignKeyDSIGN v)
+                                                  encodeSignKeyDSIGN
+                                                  decodeSignKeyDSIGN
+        , testProperty "Sig"     $ prop_cbor_with @(SigDSIGN v)
+                                                  encodeSigDSIGN
+                                                  decodeSigDSIGN
+        ]
+
+      , testGroup "To/FromCBOR class"
+        [ testProperty "VerKey"  $ prop_cbor @(VerKeyDSIGN v)
+        , testProperty "SignKey" $ prop_cbor @(SignKeyDSIGN v)
+        , testProperty "Sig"     $ prop_cbor @(SigDSIGN v)
+        ]
+
+      , testGroup "ToCBOR size"
+        [ testProperty "VerKey"  $ prop_cbor_size @(VerKeyDSIGN v)
+        , testProperty "SignKey" $ prop_cbor_size @(SignKeyDSIGN v)
+        , testProperty "Sig"     $ prop_cbor_size @(SigDSIGN v)
+        ]
+
+      , testGroup "direct matches class"
+        [ testProperty "VerKey"  $ prop_cbor_direct_vs_class @(VerKeyDSIGN v)
+                                                             encodeVerKeyDSIGN
+        , testProperty "SignKey" $ prop_cbor_direct_vs_class @(SignKeyDSIGN v)
+                                                             encodeSignKeyDSIGN
+        , testProperty "Sig"     $ prop_cbor_direct_vs_class @(SigDSIGN v)
+                                                             encodeSigDSIGN
+        ]
+      ]
+
+    , testGroup "verify"
+      [ testProperty "verify positive" $ prop_dsign_verify_pos @v
+      , testProperty "verify newgative (wrong key)" $ prop_dsign_verify_neg_key @v
+      , testProperty "verify newgative (wrong message)" $ prop_dsign_verify_neg_msg @v
```

Comment:
Should this be spelt "negative"

Reply:
Yeah, I think so, thanks. I'll fix it. 

This was copy-pasted from `cardano-base`. I'll be raising a PR on that repo to expose this whole thing as a utility function, so I'll probably remove it from this repo eventually.